### PR TITLE
Check server connection before starting client from launcher

### PR DESCRIPTION
**Pull request by robinkanters**

### CODE REVIEW

File: TrueCraft.Launcher/Views/MultiplayerView.cs [hunk 1]
```diff
@@ -131,7 +143,10 @@ public MultiplayerView(LauncherWindow window)
             foreach (var server in UserSettings.Local.FavoriteServers)
             {
                 var row = ServerListStore.AddRow();
-                using (var stream = Assembly.GetExecutingAssembly().GetManifestResourceStream("TrueCraft.Launcher.Content.default-server-icon.png"))
+                using (
+                    var stream =
+                        Assembly.GetExecutingAssembly()
+                            .GetManifestResourceStream("TrueCraft.Launcher.Content.default-server-icon.png"))
```
Comment (SirCmpwn) [inline TrueCraft.Launcher/Views/MultiplayerView.cs]: What's up with this shit?

File: TrueCraft.Launcher/Views/MultiplayerView.cs [hunk 2]
```diff
@@ -152,26 +167,43 @@ public MultiplayerView(LauncherWindow window)
...
-            this.PackStart(ServerListView);
-            this.PackStart(addServerHBox);
-            this.PackStart(ServerCreationBox);
+            PackEnd(BackButton);
+            PackEnd(ConnectButton);
+            PackStart(MultiplayerLabel);
+            PackStart(ServerIPEntry);
+            PackStart(ServerListView);
+            PackStart(addServerHBox);
+            PackStart(ServerCreationBox);
```
Comment (SirCmpwn) [inline TrueCraft.Launcher/Views/MultiplayerView.cs]: Why are you making all of these unrelated changes? This is not appropriate.

Comment (robinkanters) [inline TrueCraft.Launcher/Views/MultiplayerView.cs]: > Cleaned some code

**Comment by SirCmpwn** [comment]
Aside from the basic problems present here, this is not something I want to do this way. I want the client to return with a nonzero status code and print the reason to stderr, which the launcher can pick up.

**Comment by SirCmpwn** [comment]
Tip for next time: join IRC and talk about the changes you want to make before you do them.

**Comment by robinkanters** [comment]
> basic problems

there are none, but sure

**Comment by SirCmpwn** [comment]
"basic problems" referred to things like changing unrelated parts of the code and the weird using statements.

**Comment by robinkanters** [comment]
the first one being deliberate, see the commit message, the second is to improve readability. I don't see the problem

**Comment by SirCmpwn** [comment]
Opinionated changes like that should be asked about first.

**Comment by SirCmpwn** [comment]
Remember that you aren't the only one working on this and your personal coding style preferences don't apply here.

File: TrueCraft.Launcher/Views/MultiplayerView.cs (hunk 1 above)
Comment (robinkanters) [inline TrueCraft.Launcher/Views/MultiplayerView.cs]: Then reject those, don't base a functional decision on code style

Comment (SirCmpwn) [inline TrueCraft.Launcher/Views/MultiplayerView.cs]: Look dude, coding style will be enforced on this project. You do not get to change that. Even so, I can't reject or accept only parts of the pull request. I would have left it open and given you a chance to undo the coding style changes, but I rejected it because the way you went about implementing this feature is wrong. Take it easy, dude.

Comment (SirCmpwn) [inline TrueCraft.Launcher/Views/MultiplayerView.cs]: It's clear that it was a mistake to give you write access to this repository.

Comment (robinkanters) [inline TrueCraft.Launcher/Views/MultiplayerView.cs]: That would be clear if I pushed it, doesn't it count for something that I
created a PR? I guess you really don't get it, too bad

Comment (SirCmpwn) [inline TrueCraft.Launcher/Views/MultiplayerView.cs]: The content of this pull request isn't relevant, your attitude is.
